### ISSUE TITLE
Issues with 3.0.0.1 - n.2

### ISSUE THREAD

**Issue by jonathan-dejong** [Billmate/woocommerce#216]
Hi!

I posted here: https://wordpress.org/support/topic/3-0-0-1-file-missing/
But it seems this is the place to be instead! 

I'm super exited for the new checkout! Looking forward to testing it properly.

In addition to the things pointed out in the above link, billmate has not been updated to work with WooCommerce 3.0 and it's new CRUD methods. 3.0 has been out for 3 weeks and most people have updated or are going to (hopefully) very soon. I really hope an update will be out soon with the fixes necessary. Right now I can't even get the new checkout to work with WooCommerce 3.0. It just says `errorcode 50000`. 

Are you open for PRs if I can help and have the time?

**Comment by Boxedsolutions** [comment]
Sure PRs are fine, 

It was something that happened when pushed to wordpress repo, but should be fixed in 3.0.0.2 in Wordpress repo or just download from here and upload. 

WC 3.0 is a huge update from what i Understand. But we will look at it.

**Comment by jonathan-dejong** [comment]
@Boxedsolutions 
Yeah and I had version 3.0.0.2 but there are still issues with that code which I mention in the latter posts in the linked forum topic. 

After I changed the code according to my topic the fatal errors are no more. However the errorcode 5000 are still there. 

WC 3.0 is huge in that one has to change all interaction with WC data to the new CRUD system. 
So there's usually A LOT of places to change but fairly easy to do so... puh. We have lots of stores we're updating and the new methods are easier and better but damn it's a pain.

**Comment by Boxedsolutions** [comment]
Better to add them to Github, shall tell the staff to add a notification that issues is best posted here.

Where did errorcode 50000 come from

**Comment by ben72** [comment]
OK so the latest version of Billmate is not tested/working with WC 3.0.x? I'm not talking about the new Checkout feature.
Good to know as I was planning to update some sites the coming week.
Do you have an ETA for it?

**Comment by Boxedsolutions** [comment]
Now we have not tested with WC 3.0.

**Comment by ben72** [comment]
As an improvement suggestion I think you should add the latest version of WC you have tested with in the description in the WP repos.

Now it says in the FAQ in the WP repo for Billmate for WC:
Minimum Requirements
* Account at Billmate.se
* WooCommerce version 2.3 and higher

"and higher" is not true.

**Comment by Boxedsolutions** [comment]
Great suggestion, will implement it.

**Comment by jonathan-dejong** [comment]
Okay I'll lay it out here too.
Upon updating to 3.0.0.2 there is this issue:
`
[24-Apr-2017 18:57:46 UTC] PHP Fatal error:  Uncaught Error: Call to a member function is_empty() on null in /srv/www/private/stjarnbarn/htdocs/wp-content/plugins/billmate-payment-gateway-for-woocommerce/shortcodes.php:49
`

Which I think actually stems from the line above that:
`add_action('init','register_billmate_incomplete_order_status');`
because there is no function called `register_billmate_incomplete_order_status` anywhere.
There is however a function called `add_billmate_incomplete_order_statuses ` which in term is not called upon from anywhere. 

So I changed the hook myself to call this function which resulted in a working website and what seems to be, generally, working billmate. 
When heading to the new billmate checkout page tho, I get a 50000 error
![skarmavbild 2017-04-25 kl 12 59 40](https://cloud.githubusercontent.com/assets/1508751/25382266/1bdc9ac6-29b7-11e7-89f4-24c8e5bf9503.png)

If I check the billmate API log it seems checkout is not activated on our account:
`"message": "Billmate Error Code: 50000. Checkout: Service \u00c3\u00a4r inte aktiverat f\u00c3\u00b6r ditt konto",`

So I guess I'll also have to ask them to activate it. 

I'm gonna try checkout on WC 2.6 as well since I've only updated on my local site so far.
